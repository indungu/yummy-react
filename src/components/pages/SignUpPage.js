import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import SignUpForm from '../forms/SignUpForm';
import NavBar from '../common/NavBar';
import Footer from '../common/Footer';

export default class SignUpPage extends Component {
  render() {
    const token = window.localStorage.getItem('token');
    if (token) {
      return <Redirect to="/dashboard" />;
    }
    return (
      <div>
        <NavBar />
        <div className="">
          <div className="container w-40">
            <SignUpForm />
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
