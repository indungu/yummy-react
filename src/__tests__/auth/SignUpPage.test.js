import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SignUpPage from '../../components/pages/SignUpPage';

describe('SignUpPage', () => {
  const wrapper = shallow(<SignUpPage />);
  it('Renders properly ', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders all the divs', () => {
    expect(wrapper.find('div').length).toBe(3);
  });
});
