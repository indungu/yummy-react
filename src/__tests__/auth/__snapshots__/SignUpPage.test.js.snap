// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage Renders properly  1`] = `
<div>
  <NavBar />
  <div
    className=""
  >
    <div
      className="container w-40"
    >
      <SignUpForm />
    </div>
  </div>
  <Footer />
</div>
`;
